### ISSUE TITLE
Oauth support

### ISSUE THREAD

**Issue by serj-p** [dylex/slack-libpurple#24]
I work at Oracle and they are unfriendly with legacy token mechanism. 
The only way to use the plugin is to use the oauth token, in order to do this the plugin has to be registered as a slack official app and implement the functionality to issue a token. 

https://api.slack.com/docs/oauth#flow

I understand that this is a huge peace of work, but is there any plans to have it in feasible future?

**Comment by dylex** [comment]
I will look into this. The first step was apparently making a new slack for the plugin itself (ha! the whole point of this was to have to interact with slack less!) which I've now done ([libpurple](https://join.slack.com/t/libpurple/shared_invite/enQtMzEyNzYyODkyNDA0LTE2ZDZlYWMzMjRkYzI0YjNjMzJhODllYjBlZjk2OThhM2JmNmVhNWRlZDViY2FiNGM0MjZjMDNmZjEwMzI4NWE)). Can't promise anything, but I agree it would be nice.

I should warn you that even if this works, oracle may still have to approve use of the registered app with their slack. Not really sure, though.

**Comment by EionRobb** [comment]
Could also have an oauth receiver page on pidgin.im that can provide the token to put into the app, if that helps.

**Comment by EionRobb** [comment]
I'd still prefer to take the tokens out of one of the official apps instead

**Comment by jgoerzen** [comment]
https://github.com/wee-slack/wee-slack manages this and their approach may be instructive.

**Comment by dylex** [comment]
Indeed: https://github.com/wee-slack/wee-slack/blob/master/wee_slack.py#L3070
They publish the secret, and make users copy the key from the redirected url.
(In fact, we could just use their client id since it's already registered.)

**Comment by serj-p** [comment]
got it working copying token from this API
`https://your-org-domain.slack.com/api/apps.actions.list`
it sends params:
`token: xoxs-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx`

I used my login without @domain and set host as it is in web browser.

**Comment by 257** [comment]
> got it working copying token from this API
> `https://your-org-domain.slack.com/api/apps.actions.list`
> it sends params:
> `token: xoxs-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx`

i tried this in my browser (already logged in):
```https://timescaledb.slack.com/api/apps.actions.list```
it's telling me this:
```{"ok":false,"error":"not_authed"}```

any idea?

**Comment by dispatchrabbi** [comment]
> I am curious whether teams that have disabled legacy tokens [...] would be willing to approve an app integration like this.

Hi! I'm an admin on a team that hasn't quite disabled legacy tokens but would like to. For me, the primary advantages here are a) we get to see the scopes that the token will be used for[0], and b) our users won't get pissed when Slack finally does retire legacy API tokens.

[0] I know that Pidgin will need to use quite a large number of scopes, but definitely not all of them, and I favor explicit declaration with this kind of thing.

**Comment by grimmy** [comment]
PR #107 should work around this as long as the slack still supports the normal mobile client login.

**Comment by tino415** [comment]
Did you consider using deep link as redirect uri? I recently experimented with it in one electron app and it works flawlessly (well, at last on mac)
